### PR TITLE
[REVIEW][FEATURE][GAWE-126] - Edit Task including Sub Task

### PR DESCRIPTION
**Pull request by achmadss**
#### Description

> Edit Task including Sub Task

#### Ticket Reference(s)

[GAWE-126](https://bluehabit.atlassian.net/browse/GAWE-126)

##### UI Changes

> Screenshot or video of UI-related changes

##### Coverage

> Test coverage screenshot.

N/A

#### Note(s)

> Mention other user or extra note unrelated to the MR here.

[GAWE-126]: https://bluehabit.atlassian.net/browse/GAWE-126?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/com/bluehabit/eureka/controller/TaskController.java [hunk 1]
```diff
@@ -148,4 +149,16 @@ public ResponseEntity<BaseResponse<PageResponse<Task>>> getListTaskByStatus(
     ) {
         return taskService.getListTaskByStatus(statusId, pageable);
     }
+
+    @PostMapping(
+        path = "publish",
```
Comment (triandamai) [inline src/main/java/com/bluehabit/eureka/controller/TaskController.java]: edit task? tapi pathnya "publish" ? @achmadss

File: src/main/java/com/bluehabit/eureka/services/TaskService.java [hunk 2]
```diff
@@ -278,4 +281,52 @@ public ResponseEntity<BaseResponse<PageResponse<Task>>> getListTaskByStatus(
...
+                                    subTaskRepository.save(subTask);
+                                }
+                            });
+                        taskRepository.save(task);
+                        return BaseResponse.success(translate(""), task);
+                    })
+                    .orElseThrow(() -> new GeneralErrorException(HttpStatus.NOT_FOUND.value(), translate("")));
+
+            } catch (ParseException parseException) {
+                throw new GeneralErrorException(HttpStatus.BAD_REQUEST.value(), parseException.getMessage());
```
Comment (triandamai) [inline src/main/java/com/bluehabit/eureka/services/TaskService.java:360]: ini salah parsing yang mana ya? @achmadss

**Review by triandamai — CHANGES_REQUESTED** [review]

File: src/main/java/com/bluehabit/eureka/services/TaskService.java (hunk 2 above)
Comment (achmadss) [reply src/main/java/com/bluehabit/eureka/services/TaskService.java:360]: ttu w bingung ya, date nya itu yg dikirim dari FE bentuk nya gimana? epoch atau yyyy-MM-dd? kalo yyyy-MM-dd kan ada function parse nya, nah itu katanya hrs di surround try/catch

File: src/main/java/com/bluehabit/eureka/controller/TaskController.java (hunk 1 above)
Comment (achmadss) [reply src/main/java/com/bluehabit/eureka/controller/TaskController.java]: check 5169ba8e

**Review by triandamai — APPROVED** [review]

File: src/main/java/com/bluehabit/eureka/services/TaskService.java (hunk 2 above)
Comment (triandamai) [reply src/main/java/com/bluehabit/eureka/services/TaskService.java:360]: i see

**Pull request merged**